metrics/src/macros.rs

```diff
@@ -0,0 +1,870 @@
+#[doc(hidden)]
+#[macro_export]
+macro_rules! metadata_var {
+    ($target:expr, $level:expr) => {{
+        static METADATA: $crate::Metadata<'static> =
+            $crate::Metadata::new($target, $level, ::core::option::Option::Some(module_path!()));
+        &METADATA
+    }};
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! count {
+    () => {
+        0usize
+    };
+    ($head:tt $($tail:tt)*) => {
+        1usize + $crate::count!($($tail)*)
+    };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! key_var {
+    ($name: literal) => {{
+        static METRIC_KEY: $crate::Key = $crate::Key::from_static_name($name);
+        &METRIC_KEY
+    }};
+    ($name:expr) => {
+        $crate::Key::from_name($name)
+    };
+    ($name:literal, $($label_key:literal => $label_value:literal),*) => {{
+        static LABELS: [$crate::Label; $crate::count!($($label_key)*)] = [
+            $($crate::Label::from_static_parts(&$label_key, &$label_value)),*
+        ];
+        static METRIC_KEY: $crate::Key = $crate::Key::from_static_parts($name, &LABELS);
+        &METRIC_KEY
+    }};
+    ($name:expr, $($label_key:literal => $label_value:literal),*) => {{
+        static LABELS: [$crate::Label; $crate::count!($($label_key)*)] = [
+            $($crate::Label::from_static_parts($label_key, $label_value)),*
+        ];
+        $crate::Key::from_static_labels($name, &LABELS)
+    }};
+    ($name:expr, $($label_key:expr => $label_value:expr),*) => {{
+        let labels = ::std::vec![
+            $($crate::Label::new($label_key, $label_value)),*
+        ];
+        $crate::Key::from_parts($name, labels)
+    }};
+    ($name:expr, $labels:expr) => {
+        $crate::Key::from_parts($name, $labels)
+    }
+}
+
+/// Registers a counter.
+///
+/// Counters represent a single monotonic value, which means the value can only be incremented, not
+/// decremented, and always starts out with an initial value of zero.
+///
+/// Metrics can be registered, which provides a handle to directly update that metric.  For
+/// counters, [`Counter`](crate:Counter) is provided which can be incremented or set to an absolute value.
```

Comment:
This should be `crate::Counter`.

Reply:
02f230f25d4b6b3ee0074fd8ee0fbd60a075dd26